internal/rbd/controllerserver.go

```diff
@@ -786,16 +784,15 @@ func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 
 	rbdVol, err := genVolFromVolID(ctx, volID, cr, req.GetSecrets())
 	if err != nil {
-		if _, ok := err.(ErrImageNotFound); ok {
+		switch err.(type) {
+		case *ErrImageNotFound:
 			return nil, status.Errorf(codes.NotFound, "volume ID %s not found", volID)
-		}
-
-		if _, ok := err.(util.ErrPoolNotFound); ok {
+		case *util.ErrPoolNotFound:
 			klog.Errorf(util.Log(ctx, "failed to get backend volume for %s: %v"), volID, err)
 			return nil, status.Errorf(codes.InvalidArgument, err.Error())
+		default:
+			return nil, status.Errorf(codes.Internal, err.Error())
 		}
```

Comment:
```suggestion
switch err.(type) {
		case *ErrImageNotFound:
			err=status.Errorf(codes.NotFound, "source Volume ID %s not found", req.GetSourceVolumeId())
		case *util.ErrPoolNotFound:
			klog.Errorf(util.Log(ctx, "failed to get backend volume for %s: %v"), req.GetSourceVolumeId(), err)
			fallthrough
		default:
			err=status.Errorf(codes.Internal, err.Error())
		}
		return nil,err
```

Reply:
this can be applied to https://github.com/ceph/ceph-csi/pull/1032/files#diff-f010f466063f62f823361f972d08b554R513 to avoid multiple returns

Reply:
Sure, I do not like all those return statements either :+1:

Reply:
Except:
```
internal/rbd/controllerserver.go:520:4: cannot fallthrough in type switch
```

So, no `fallthrough` but multiple `err` assignments.

Reply:
Yeah as you are comparing with type the fall through won't work, we can remove it